lib/builder.ml

```diff
@@ -553,19 +553,19 @@ let read fd =
     in
     let bl = Bytes.create 8 in
     let* () = r bl 8 in
-    let l = Cstruct.BE.get_uint64 (Cstruct.of_bytes bl) 0 in
+    let l = Bytes.get_int64_be bl 0 in
     let l_int = Int64.to_int l in (* TODO *)
```

Comment:
This TODO is a bit unclear to me. Is it because we don't check for overflow?

We can defer this to not lose focus as this is unchanged.

Reply:
yes, that's due to potential overflow, esp. on 32 bit platforms.